kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/KilkariServiceImpl.java

```diff
@@ -20,12 +23,22 @@
 
     @Override
     public List<SubscriptionPack> getSubscriberPacks(String callingNumber) {
+        Set<Subscription> subscriptions = getSubscriptions(callingNumber);
+        List<SubscriptionPack> subscriptionPacksForSubscriber = new ArrayList<>();
+
+        for (Subscription subscription : subscriptions) {
+            subscriptionPacksForSubscriber.add(subscription.getSubscriptionPack());
+        }
+        return subscriptionPacksForSubscriber;
+    }
+
+    @Override
+    public Set<Subscription> getSubscriptions(String callingNumber) {
         Subscriber subscriber = subscriberDataService.findByCallingNumber(callingNumber);
         if (null == subscriber) {
             //todo: handle that better (ie: proper exception?) such that the proper bad response is returned
             throw new IllegalStateException("NULL subscriber");
```

Comment:
yeah, I was thinking about just removing it and moving the retrieval of packs up to Frank's code...